Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +424,253 @@ export class CppSettings extends Settings {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundAssignmentOperator")!;
+    // Returns the value of a setting as a number with proper type validation and checks if value falls within the specified range.
+    private getAsNumber(settingName: string): number;
+    private getAsNumber(settingName: string, allowNull: boolean): number | null;
+    private getAsNumber(settingName: string, allowNull: boolean = false): number | null {
+        const value: any = super.Section.get<any>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        // Validates the value is a number and clamps it to the specified range. Allows for undefined maximum or minimum values.
+        if (isNumber(value)) {
+            if (setting.minimum !== undefined && value < setting.minimum) {
+                return setting.minimum;
+            }
+            if (setting.maximum !== undefined && value > setting.maximum) {
+                return setting.maximum;
+            }
+            return value;
+        }
+        return setting.default;
+    }
+
+    // Returns the value of a setting as an array of strings with proper type validation and checks for valid enum values.
+    private getAsArrayOfStrings(settingName: string, allowUndefinedEnums: boolean = false): string[] {
+        const value: any = super.Section.get(settingName);
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (isArrayOfString(value)) {
+            if (!allowUndefinedEnums) {
+                if (setting.items.enum !== undefined) {
+                    if (!value.every(x => this.isValidEnum(setting.items.enum, x))) {
+                        return setting.default;
+                    }
+                }
+            }
+            return value;
+        }
+        return setting.default;
     }
 
-    public get vcFormatSpacePointerReferenceAlignment(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.pointerReferenceAlignment")!;
+    private getAsExcludes(settingName: string): Excludes;
+    private getAsExcludes(settingName: string, allowNull: boolean): Excludes | null;
+    private getAsExcludes(settingName: string, allowNull: boolean = false): Excludes | null {
+        const value: any = super.Section.get(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        if (isValidMapping(value, (key) => typeof key === 'string', (val) => typeof val === 'boolean')) {
+            return value as Excludes;
```

Comment:
You fixed `isValidMapping`, but you didn't fix the value validation.  This can be a boolean or `{ when: string }`.

Also, you should be using the common helpers for the intrinsic types `isString`/`isBoolean`.   The key validator could just pass in `isString` directly. You wouldn't have to wrap that in another lambda.

Reply:
Please use `isBoolean` for the boolean part of the type check.